### PR TITLE
DAOS-7128 test: Improve NLT/junit handling.

### PR DESCRIPTION
**Pull request by ashleypittman**
Keep a valid junit file with a failure saved at all times,
then on successful exit remove the failure and re-write
the file.  Previously this was done the other way round,
however on failure the code is called during shutdown
and junit-xml is unable to write the file.

Add some debugging.

Parallel-build: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: utils/docker_nlt.sh [hunk 1]
```diff
@@ -10,4 +10,7 @@ set -e
 
 ./utils/setup_daos_admin.sh
 
+# Needed for Ubuntu
+export LD_LIBRARY_PATH=/opt/daos/prereq/release/spdk/lib/
```
Comment (ashleypittman) [inline utils/docker_nlt.sh:14]: @jolivier23 This looks to be a build/scons issue

File: utils/node_local_test.py [hunk 2]
```diff
@@ -2707,7 +2731,10 @@ def main():
     setup_log_test(conf)
 
     server = DaosServer(conf, test_class='first')
-    server.start()
+    clean=True
+    if args.no_root:
+        clean=False
+    server.start(clean=clean)
```
Comment (ashleypittman) [inline utils/node_local_test.py:2737]: This seems to be the crux of the fix although I know not why.  Without it then the server is restarting the engine a number of times.

**Comment by ashleypittman** [comment]
This resolves the previous problem, adds some more debugging and re-enables NLT on the docker builders.

**Review by tanabarr — APPROVED** [review]
looks okay but probably want review from someone more familiar with build like @brianjmurrell

File: utils/docker_nlt.sh (hunk 1 above)
Comment (jolivier23) [reply utils/docker_nlt.sh:14]: Hmm, ok.   Someone not adding a proper dependence probably.

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**